src/Identity/Controllers/SsoController.cs

```diff
@@ -60,12 +64,17 @@ public async Task<IActionResult> PreValidate(string domainHint)
                 var requestPath = $"/Account/PreValidate?domainHint={domainHint}&culture={culture}";
                 var httpClient = _clientFactory.CreateClient("InternalSso");
                 using var responseMessage = await httpClient.GetAsync(requestPath);
+
                 if (responseMessage.IsSuccessStatusCode)
                 {
-                    // All is good!
-                    return new EmptyResult();
+                    var res = responseMessage.Content.ReadAsStringAsync().Result;
+                    var obj = JsonConvert.DeserializeObject<ExpandoObject>(res, new ExpandoObjectConverter());
+
+                    return Json(obj);
                 }
+
                 Response.StatusCode = (int)responseMessage.StatusCode;
+
                 var responseJson = await responseMessage.Content.ReadAsStringAsync();
```

Comment:
It would seem that either way if we need the `responseJson` read from the response message content... we should have this code only once, higher up in the logic chain, then depending on the status code being successful or not would determine what we do with the response message.